pkg/workload/ycsb/ycsb.go

```diff
@@ -720,6 +736,7 @@ func (yw *ycsbWorker) nextReadKey() string {
 	// distribution, so it might be worthwhile to exactly emulate what they're
 	// doing.
 	rowIndex := yw.requestGen.Uint64() % rowCount
+	rowIndex += uint64(yw.config.insertStart)
```

Comment:
Missing `% rowCount`, otherwise `rowIndex` may be out of bounds. This whole scheme is also questionable; e.g., if `requestGen` is uniform, you don't get a uniform distribution over `[insertStart, rowCount)` since the modulus isn't (necessarily) relatively prime.

Reply:
Oops, you're right. I can't remember exactly what threw me off. Typically, when seeking `rand` in a specific range, [min, max), the pattern is `rand() % (max-min+1) + min`; I suppose that's probably what misled me. While the above code is equivalent, the below is more idiomatic, imo,

```
rowIndex := yw.requestGen.Uint64() % (yw.rowCounter.Last() - yw.config.insertStart) + yw.config.insertStart
```